addons/udes_stock_cron/tests/test_stock_picking.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+from odoo.addons.udes_stock.tests.common import BaseUDES
+from odoo.exceptions import UserError
+
+
+class TestStockPicking(BaseUDES):
+    @classmethod
+    def setUpClass(cls):
+        super(TestStockPicking, cls).setUpClass()
+
+        products_info = [{"product": cls.apple, "uom_qty": 10.0}]
+
+        cls.picking_type_pick.u_num_reservable_pickings = -1
+
+        # Create a picking with a move
+        cls.test_picking_pick = cls.create_picking(
+            cls.picking_type_pick,
+            products_info=products_info,
+            confirm=True,
+            assign=False,
+            location_dest_id=cls.test_received_location_01.id,
+        )
+
+        # Create an available quantity of apples
+        cls.test_quant_apple = cls.create_quant(
+            product_id=cls.apple.id,
+            location_id=cls.picking_type_pick.default_location_src_id.id,
+            quantity=10.0,
+        )
+
+    # Want to inherit the methods from BaseUDES but do not want the database to
+    # rollback after each test, as the method in test: reserve_stock, commits to
+    # the database via the cursor - so will not be able to rollback
+    def setUp(cls):
+        pass
+
+    def tearDown(cls):
+        pass
```

Comment:
Does this mean that the records created in the test cases persist in the DB after running the tests?

Reply:
The commit in the reserve stock method means that the database can't roll back to before the tests were run. I overwrote tearDown because it tries to roll back after every test case. There is also a tearDownClass which is meant to roll back the database after all the tests are done, however it can't rollback properly due to the commit in the reserve stock method. As it stands there will still be records in the database after running the tests. I thought that maybe I could extend tearDownClass for these tests and make it drop the database?

Reply:
Until we think of a better way, let's extend `tearDownClass` to `unlink` the records created as part of these tests

Reply:
As discussed, we've found a better way through the use of `enter_test_mode` and `leave_test_mode`